org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/impl/AssistParser.java

```diff
@@ -376,37 +378,48 @@ public RecoveredElement buildInitialRecoveryState(){
 	// This is  copy of the code that processes astStack early in this method
 	for (int i = 0; i <= this.expressionPtr; i++, lastNode = node) {
 		node = this.expressionStack[i];
-		if (node == null || !(node instanceof InstanceOfExpression)) continue;
-		/* check for intermediate block creation, so recovery can properly close them afterwards */
-		int nodeStart = node.sourceStart;
-		for (int j = blockIndex; j <= this.realBlockPtr; j++){
-			if (this.blockStarts[j] >= 0) {
-				if (this.blockStarts[j] > nodeStart){
-					blockIndex = j; // shift the index to the new block
-					break;
-				}
-				if (this.blockStarts[j] != lastStart){ // avoid multiple block if at same position
+		if (node instanceof InstanceOfExpression) {
```

Comment:
Please review the changes while ignoring empty space changes. The block in this `if` has new indentation.